site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx

```diff
@@ -113,172 +121,184 @@ export const CreateWorkspacePageView: FC<CreateWorkspacePageViewProps> = ({
-                      name: parameter.name,
-                      value: value,
-                    });
-                  },
-                  disabled:
-                    disabledParamsList?.includes(
-                      parameter.name.toLowerCase().replace(/ /g, "_"),
-                    ) || form.isSubmitting,
-                };
-              }}
-            />
-          </>
-        )}
+          {parameters && (
+            <>
+              <MutableTemplateParametersSection
+                templateParameters={parameters}
+                getInputProps={(parameter, index) => {
+                  return {
+                    ...getFieldHelpers(
+                      "rich_parameter_values[" + index + "].value",
+                    ),
+                    onChange: async (value) => {
+                      await form.setFieldValue(
+                        "rich_parameter_values." + index,
+                        {
+                          name: parameter.name,
+                          value: value,
+                        },
+                      );
+                    },
+                    disabled:
+                      disabledParamsList?.includes(
+                        parameter.name.toLowerCase().replace(/ /g, "_"),
+                      ) || form.isSubmitting,
+                  };
+                }}
+              />
+              <ImmutableTemplateParametersSection
+                templateParameters={parameters}
+                classes={{ root: styles.warningSection }}
+                getInputProps={(parameter, index) => {
+                  return {
+                    ...getFieldHelpers(
+                      "rich_parameter_values[" + index + "].value",
+                    ),
+                    onChange: async (value) => {
+                      await form.setFieldValue(
+                        "rich_parameter_values." + index,
+                        {
+                          name: parameter.name,
+                          value: value,
+                        },
+                      );
+                    },
+                    disabled:
+                      disabledParamsList?.includes(
+                        parameter.name.toLowerCase().replace(/ /g, "_"),
+                      ) || form.isSubmitting,
+                  };
+                }}
+              />
+            </>
+          )}
 
-        <FormFooter
-          onCancel={onCancel}
-          isLoading={creatingWorkspace}
-          submitLabel="Create Workspace"
-        />
-      </HorizontalForm>
-    </FullPageHorizontalForm>
+          <FormFooter
+            onCancel={onCancel}
+            isLoading={creatingWorkspace}
+            submitLabel="Create Workspace"
+          />
+        </HorizontalForm>
+      </FullPageHorizontalForm>
+    </>
   );
 };
 
-type ExternalAuthErrors = Record<string, string>;
+function getAuthErrors(
```

Comment:
This replaces the `useExternalAuthVerification` component. Everything inside can be derived from existing state, and introducing more state means more things that can potentially get out of sync